### PR TITLE
feat: if  article hasn't been updated don't publish an update

### LINKED ISSUE #6: if article is the same do not update
As this can be used as a mono repo for articles, every time one of them if going to change, they'll all get published.

Thanks to https://github.com/thepracticaldev/dev.to/pull/3394 we can now fetch an article and its markdown so we should be able to compare the local markdown with the remote one and skip the update if there's no diff.

### PR DESCRIPTION
**Pull request by maxime1992**
This will avoid to update the date of latest update on an article while publishing others

Closes maxime1992/dev-to-git#6

### CODE REVIEW

**Comment by maxime1992** [comment]
Manually tested with my latest article, output is:

> Article "Enigma machine, how does the famous encryption device work?" is already up to date

:smile:

File: src/article.ts [hunk 1]
```diff
@@ -69,6 +78,16 @@ export class Article {
       });
     }
 
+    const remoteArticleBodyMarkdown: string | null = await this.fetchArticleBodyMarkdown(this.articleConfig.id);
+
+    if (!!remoteArticleBodyMarkdown && remoteArticleBodyMarkdown.trim() === body.body_markdown.trim()) {
+      return Promise.resolve({
```
Comment (zakhenry) [inline src/article.ts]: This function is already async right? Can’t you just return the object?

File: src/article.ts [hunk 2]
```diff
@@ -53,7 +53,16 @@ export class Article {
...
   }
 
-  public publishArticle(token: string): Promise<ArticlePublishedStatus> {
+  private fetchArticleBodyMarkdown(articleId: number): Promise<string | null> {
+    return got(`https://dev.to/api/articles/${articleId}`, {
+      json: true,
+      method: 'GET',
+    })
+      .then((x: got.Response<ArticleApi>) => x.body.body_markdown)
+      .catch(() => null);
```
Comment (zakhenry) [inline src/article.ts]: Should at least console.error here, but why not throw?

**Review by zakhenry — CHANGES_REQUESTED** [review]

File: src/article.ts (hunk 1 above)
Comment (maxime1992) [reply src/article.ts]: Done

File: src/article.ts (hunk 2 above)
Comment (maxime1992) [reply src/article.ts]: Agree. Fixed

**Review by maxime1992 — COMMENTED** [review]
@zakhenry I've fixed everything and slightly refactored the way I handled the errors to display them when failing and also exiting the process with an error code so that on CI it doesn't silently passes

File: src/dev-to-git.ts [hunk 3]
```diff
@@ -120,10 +121,22 @@ export class DevToGit {
...
-  .then(statuses => devToGit.logger(statuses))
-  .catch(err => {
+  .then(articles => ({ articles, text: formatArticlePublishedStatuses(articles) }))
+  .then(res => {
+    devToGit.logger(res.text);
+
+    res.articles.forEach(article => {
+      if (
+        article.updateStatus === UpdateStatus.ERROR ||
+        article.updateStatus === UpdateStatus.FAILED_TO_EXTRACT_FRONT_MATTER
```
Comment (zakhenry) [inline src/dev-to-git.ts:131]: prob followup, but this is a very good case for a switch case, with the default being an error

**Review by zakhenry — APPROVED** [review]

**Pull request merged**